x-pack/filebeat/input/awscloudwatch/input.go

```diff
@@ -120,36 +118,54 @@ func (in *cloudwatchInput) Run(inputContext v2.Context, pipeline beat.Pipeline)
 	cwPoller := newCloudwatchPoller(
 		log.Named("cloudwatch_poller"),
 		in.metrics,
-		in.awsConfig.Region,
+		region,
 		in.config)
 	logProcessor := newLogProcessor(log.Named("log_processor"), in.metrics, client, ctx)
-	cwPoller.metrics.logGroupsTotal.Add(uint64(len(logGroupNames)))
+	cwPoller.metrics.logGroupsTotal.Add(uint64(len(logGroupIDs)))
 	cwPoller.startWorkers(ctx, svc, logProcessor)
-	cwPoller.receive(ctx, logGroupNames, time.Now)
+	cwPoller.receive(ctx, logGroupIDs, time.Now)
 	return nil
 }
 
-func parseARN(logGroupARN string) (string, string, error) {
-	arnParsed, err := arn.Parse(logGroupARN)
-	if err != nil {
-		return "", "", fmt.Errorf("error Parse arn %s: %w", logGroupARN, err)
-	}
+// fromConfig is a helper to parse input configurations and derive logGroupIDs & aws region
+// Returned logGroupIDs could be empty, which require other fallback mechanisms to derive them.
+// See getLogGroupNames for example.
+func fromConfig(cfg config, awsCfg awssdk.Config) (logGrouIDs []string, region string, err error) {
+	// LogGroupARN has precedence over LogGroupName & RegionName
+	if cfg.LogGroupARN != "" {
+		parsedArn, err := arn.Parse(cfg.LogGroupARN)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to parse log group ARN: %w", err)
+		}
 
-	if strings.Contains(arnParsed.Resource, ":") {
-		resourceARNSplit := strings.Split(arnParsed.Resource, ":")
-		if len(resourceARNSplit) >= 2 && resourceARNSplit[0] == "log-group" {
-			return resourceARNSplit[1], arnParsed.Region, nil
+		if parsedArn.Region == "" {
+			return nil, "", fmt.Errorf("failed to parse log group ARN: missing region")
 		}
+
+		// refine to match AWS API parameter regex of logGroupIdentifier
+		groupId := strings.TrimSuffix(cfg.LogGroupARN, ":*")
+		logGrouIDs = append(logGrouIDs, groupId)
```

Comment:
```suggestion
		logGroupIDs = append(logGroupIDs, groupId)
```

Reply:
fixed in the latest commit - https://github.com/elastic/beats/pull/41188/commits/e8f2ef9fbe3cc8e1007172d07c594d4b97f7fcec